scripts/slack_notification_action.go

```diff
@@ -0,0 +1,246 @@
+		mainFields := []Field{
+			{
+				Title: os.Getenv(EnvSlackTitle),
+				Value: envOr(EnvSlackMessage, "EOM"),
+				Short: false,
+			},
+		}
+		fields = append(mainFields, fields...)
+	} else if minimal != "" {
+		requiredFields := strings.Split(minimal, ",")
+		mainFields := []Field{
+			{
+				Title: os.Getenv(EnvSlackTitle),
+				Value: envOr(EnvSlackMessage, "EOM"),
+				Short: false,
+			},
+		}
+		for _, requiredField := range requiredFields {
+			switch strings.ToLower(requiredField) {
+			case "ref":
+				field := []Field{
+					{
+						Title: "Ref",
+						Value: os.Getenv("GITHUB_REF"),
+						Short: true,
+					},
+				}
+				mainFields = append(field, mainFields...)
+			case "event":
+				field := []Field{
+					{
+						Title: "Event",
+						Value: os.Getenv("GITHUB_EVENT_NAME"),
+						Short: true,
+					},
+				}
+				mainFields = append(field, mainFields...)
+			case "actions url":
+				field := []Field{
+					{
+						Title: "Actions URL",
+						Value: "<" + os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv("GITHUB_REPOSITORY") + "/commit/" + os.Getenv("GITHUB_SHA") + "/checks|" + os.Getenv("GITHUB_WORKFLOW") + ">",
+						Short: true,
+					},
+				}
+				mainFields = append(field, mainFields...)
+			case "commit":
+				field := []Field{
+					{
+						Title: "Commit",
+						Value: "<" + os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv("GITHUB_REPOSITORY") + "/commit/" + os.Getenv("GITHUB_SHA") + "|" + commit_sha + ">",
+						Short: true,
+					},
+				}
+				mainFields = append(field, mainFields...)
+			}
+		}
+		fields = append(mainFields, fields...)
+	} else {
+		mainFields := []Field{
+			{
+				Title: "Ref",
+				Value: os.Getenv("GITHUB_REF"),
+				Short: true,
+			}, {
+				Title: "Event",
+				Value: os.Getenv("GITHUB_EVENT_NAME"),
+				Short: true,
+			},
+			{
+				Title: "Actions URL",
+				Value: "<" + os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv("GITHUB_REPOSITORY") + "/commit/" + os.Getenv("GITHUB_SHA") + "/checks|" + os.Getenv("GITHUB_WORKFLOW") + ">",
+				Short: true,
+			},
+			{
+				Title: "Commit",
+				Value: "<" + os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv("GITHUB_REPOSITORY") + "/commit/" + os.Getenv("GITHUB_SHA") + "|" + commit_sha + ">",
+				Short: true,
+			},
+			{
+				Title: os.Getenv(EnvSlackTitle),
+				Value: envOr(EnvSlackMessage, "EOM"),
+				Short: false,
+			},
+		}
+		fields = append(mainFields, fields...)
+	}
+
+	hostName := os.Getenv(EnvHostName)
+	if hostName != "" {
+		newfields := []Field{
+			{
+				Title: os.Getenv("SITE_TITLE"),
+				Value: os.Getenv(EnvSiteName),
+				Short: true,
+			},
+			{
+				Title: os.Getenv("HOST_TITLE"),
+				Value: os.Getenv(EnvHostName),
+				Short: true,
+			},
+		}
+		fields = append(newfields, fields...)
+	}
+
+	color := ""
+	switch os.Getenv(EnvSlackColor) {
+	case "success":
+		color = "good"
+	case "cancelled":
+		color = "#808080"
+	case "failure":
+		color = "danger"
+	default:
+		color = envOr(EnvSlackColor, "good")
```

Comment:
Should the default really be "good"? Isn't this what we would see if we have some unexpected outcome (infrastructure unavailable?)... I would assume we'd like to see some warning or error, not that it worked.

Reply:
I'll see if I can find one that's a neutral colour (say, yellow or grey)

Reply:
Or failure. We should assume the "bad" path here, not the "good" path.